internal/rules/mechanisms/authenticators/jwt_authenticator.go

```diff
@@ -91,16 +100,24 @@ func newJwtAuthenticator(id string, rawConfig map[string]any) (*jwtAuthenticator
 		return nil, err
 	}
 
-	if conf.Endpoint.Headers == nil {
-		conf.Endpoint.Headers = make(map[string]string)
+	// Pick the right endpoint to apply defaults to, JWKSEndpoint if set, otherwise the discovery endpoint
+	endpoint := x.IfThenElse(conf.JWKSEndpoint != nil, conf.JWKSEndpoint, conf.DiscoveryEndpoint)
+
+	if endpoint.Headers == nil {
+		endpoint.Headers = make(map[string]string)
+	}
+
+	if _, ok := endpoint.Headers["Accept-Type"]; !ok {
+		endpoint.Headers["Accept-Type"] = "application/json"
 	}
 
-	if _, ok := conf.Endpoint.Headers["Accept-Type"]; !ok {
-		conf.Endpoint.Headers["Accept-Type"] = "application/json"
+	if len(endpoint.Method) == 0 {
+		endpoint.Method = "GET"
 	}
 
-	if len(conf.Endpoint.Method) == 0 {
-		conf.Endpoint.Method = "GET"
+	if conf.JWKSEndpoint != nil && len(conf.Assertions.TrustedIssuers) == 0 {
```

Comment:
Is it possible to cover that with the `validate` tag?

Reply:
as far as I know there's no such thing in validator, since it's not possible to change the sub struct validation based on the outer struct.

Reply:
There is way to e.g. disable the validation with `validate:"-"` (https://github.com/dadrus/heimdall/blob/04b106631d8db2e22b1f8d167091ab2e52eb28c2/internal/rules/mechanisms/authenticators/jwt_authenticator.go#L190). Maybe a combination, like you did in line 87 is possible (Just an idea).

Reply:
What i could see: registering a custom validation tag and hooking into validator

Reply:
Ok. If no built-in option is available, we can postpone that after the relevant functionality is in place. Maybe even to a separate PR (as you want ;) ).